cilium/CFP-33716-distributed-packet-tracing.md

```diff
@@ -0,0 +1,586 @@
+# CFP-2024-06-12: Packet Tracing
+
+## Meta
+**SIG:**  
+**Sharing:** Public  
+**Begin Design Discussion:** Jun 12, 2024  
+**End Design Discussion:**  
+**Cilium Release:**  
+**Authors:** Ben Bigdelle <bigdelle@google.com>, Mark St. John <markstjohn@google.com> based on the work of Nathan Perkins <nathanperkins@google.com>  
+**Reviewers:**  
+
+## Summary
+We aim to enhance cross-cluster and multi-system observability for flows that contain an `ip_trace_id` IP Option (referred to as a tagged flow), thus supporting tracing across networks. These tagged flows will trigger adding the ID value to Flow events messages and client (Hubble / Cilium Monitor) representation. This allows correlating flow events across systems which integrate with the span observability.
+
+Additionally, within Cilium, these flows will generate datapath debugging events regardless of the global debug-verbose setting. This will improve debuggability of production systems.
+
+## Motivation
+There are currently limitations for gathering and correlating flow events:
+- NAT transformations and DSR can all change the 4-tuple, so that a single Hubble or tcpdump filter cannot be used to trace the packet along the entire path – even when using a system such as Hubble Relay.
+- Enabling verbose datapath debugging is not possible in production environments due to traffic volume which leads to dropping flows.
+
+## Goals
+- For tagged packets, add the ID to emitted flow events.
```

Comment:
Somehow this is less clear for me, suggest:

- Enable dataplane to watch for `ip_trace_id` option tagged packets and emit flow events for tagged packets.

Reply:
Fixed.